src/Descriptor.ts

```diff
@@ -0,0 +1,62 @@
+enum DescriptorKind {
+  Package,
+  Type,
+  Term,
+  Meta,
+  Method,
+  Parameter,
+  TypeParameter,
+}
+
+export class Descriptor {
+  private constructor(
+    public readonly name: string,
+    public readonly kind: DescriptorKind,
+    public readonly disambiguator?: string
+  ) {}
+  public static package(name: string): Descriptor {
+    return new Descriptor(name, DescriptorKind.Package)
```

Comment:
Should this be `module` (or `module_`) instead of `package`? (And similarly for `DescriptorKind.Package`) It looks like this is being used for modules, so it's a little confusing because there is another `static package` which is indeed being used for a package (there's a version involved).